### ISSUE TITLE
Modal dialog is too easy to bypass

### ISSUE THREAD

**Issue by joaomoreno** [microsoft/vscode#122520]
Testing https://github.com/microsoft/vscode/issues/122252

1. Have custom title bar

🐛

![recording (11)](https://user-images.githubusercontent.com/22350/116395779-6a69ea00-a7d9-11eb-920e-4b5d39790c48.gif)

**Comment by joaomoreno** [comment]
Are we (should we) be using the [`<dialog>` HTML element](https://alligator.io/html/dialog-element/)?

**Comment by sbatten** [comment]
no https://caniuse.com/?search=dialog

**Comment by bpasero** [comment]
> no https://caniuse.com/?search=dialog

Well, we could do browser detection and still use `dialog` where supported to get the best experience. I would also think that using `dialog` elements has some impact on accessibility / screen readers.

**Comment by joaomoreno** [comment]
![recording (22)](https://user-images.githubusercontent.com/22350/116523565-38b15b80-a88b-11eb-8708-e61226780913.gif)

**Comment by sbatten** [comment]
There's some trickiness with the alt listener because both event handlers use capture on the same element. I have a fix but I don't want to check it in this late in case there are conflicts with the other consumer (cc @isidorn)

**Comment by joaomoreno** [comment]
![recording (2)](https://user-images.githubusercontent.com/22350/120188078-84c52800-c1ca-11eb-93bd-5bb210d61af3.gif)

**Comment by sbatten** [comment]
I'm not sure why I got lucky before with the timing, but the preventDefault just doesn't work for this now. You can bypass the dialog, but at least its not just clicking. Can try something better next week.

**Comment by sbatten** [comment]
The menubar uses capture because it needs to function "natively" intercepting mnemonics everywhere. the dialog does the same and unfortunately is always going to add its event listener after, so it doesn't have a chance to intercept. I spent some time trying to come up with something but nothing I came up with didn't involve one component having to know about another. as I stated above, the initial "too easy" bypass was clicking which is no longer an option. since we are using a custom dialog, we will not get a truly blocking experience without some more major work that I don't see a need to prioritise.
